app/src/main/java/com/dsu/poc/ui/login/LoginFragment.kt

```diff
@@ -1,6 +1,47 @@
 package com.dsu.poc.project.ui.login
 
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.databinding.DataBindingUtil
+import androidx.lifecycle.Observer
+import androidx.lifecycle.ViewModelProviders
+import com.dsu.poc.R
 import com.dsu.poc.base.BaseFragment
+import com.dsu.poc.databinding.LoginFragmentBinding
+
+class LoginFragment : BaseFragment(){  lateinit var loginView: View
+    lateinit var loginViewModel: LoginViewModel
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        loginViewModel = ViewModelProviders.of(activity!!).get(LoginViewModel::class.java)
+    }
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
+        val loginBinder : LoginFragmentBinding = DataBindingUtil.inflate(inflater,
+            R.layout.main_login,container,false)
+        loginView = loginBinder.getRoot()
+        loginBinder.loginbtn.setOnClickListener(View.OnClickListener {
```

Comment:
use lambda functions of kotlin for click listeners.

Reply:
Click listeners should be moved to separate function like initListners() to achieve code segregation and readability